packages/htmlbars-runtime/lib/hooks.js

```diff
@@ -422,15 +422,18 @@ export function updateLocal(env, scope, name, value) {
   @param {Environment} env
   @param {Scope} scope
   @param {Function} block
+  @param {String} blockname
 
   Corresponds to entering a shadow template that was invoked by a block helper with
   `yieldIn`.
 
-  This hook is invoked with an opaque block that will be passed along to the
-  shadow template, and inserted into the shadow template when `{{yield}}` is used.
+  This hook is invoked with an opaque block that will be passed along
+  to the shadow template, and inserted into the shadow template when
+  `{{yield}}` is used. Optionally provide a non-default block name
+  that can be targeted by `{{yield to=blockName}}`.
 */
-export function bindBlock(env, scope, block) {
-  scope.block = block;
+export function bindBlock(env, scope, block, blockName='block') {
+  scope[blockName] = block;
```

Comment:
I think that he means:

```javascript
scope.blocks = scope.blocks || {};
scope.blocks[blockName] = block;
```

So that the blocks (with potentially arbitrary names in the future) don't live directly off of `scope`.